### PR TITLE
github-ci: author check github action - v10

### PR DESCRIPTION
**Pull request by jasonish**
New draft for testing as I can't test this within the scope of my own repo (or the repo that originates a pull request).

### CODE REVIEW

**Comment by jasonish** [comment]
@victorjulien This is re-worked to use raw git commands instead of the Git checkout action. I think the only way to really test is to add it to some staging pull requests, where the incorrect results showed up before.

**Comment by jasonish** [comment]
Ok, done futzing around with this version of the PR. Unfortunately, its not simple for a GitHub action in pull-request context to add a comment to its pull-request.

File: .github/workflows/authors.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        run: git log --format="%an <%ae>" origin/${GITHUB_BASE_REF}... | sort | uniq > ../commit-authors.txt
+        working-directory: suricata
+      - name: Check new authors
+        run: |
+          touch new-authors.txt
+          while read -r author; do
+             echo "Checking author: ${author}"
+             if ! grep -q "^${author}\$" authors.txt; then
+                 echo "ERROR: ${author} NOT FOUND"
+                 echo "::error ::New author found: ${author}"
```
Comment (victorjulien) [inline .github/workflows/authors.yml:32]: what happens if you use `::warning` here and then not `exit 1`?

Comment (jasonish) [reply .github/workflows/authors.yml:32]: Running 2 tests with exit 1 removed:
- https://github.com/OISF/suricata/pull/8848 (::error)
- https://github.com/OISF/suricata/pull/8849 (::warning)

I believe the exit is the only way to see any notification on the main pull request page, without going to the summary and looking for annotations.

Comment (victorjulien) [reply .github/workflows/authors.yml:32]: yeah I see, thanks

**Comment by jasonish** [comment]
Here's info on creating a comment from a pull request which might be the most useful: https://stackoverflow.com/questions/69499645/how-to-securely-allow-github-actions-to-check-pr-and-post-results-in-comment/71366152#71366152

Note that it requires getting some workflows into the master branch first, so it will require a couple of stages to get working likely. But can test most of it in my own fork, *I think*.

**Comment by jasonish** [comment]
I guess the question is, what type of output is desired:
- A red X beside the job check for others..
- OR.. A green X, but a comment specifying that there are new authors?

**Comment by victorjulien** [comment]
The latter, I think. I don't like it that  CI fails for just having new author.

**Comment by jasonish** [comment]
Continued here: https://github.com/OISF/suricata/pull/8854

**Pull request closed without merging**